### ISSUE TITLE
[Pressable] android_ripple / color doesn't work correctly

### ISSUE THREAD

**Issue by freeboub** [software-mansion/react-native-gesture-handler#3246]
### Description

on `Pressable` android_ripple={color: XXX} doesn't work correctly.

Here are issues:
- when user configure the  android_ripple={color: "#FF0000"} => the ripple color is grey. What ever the configured color is, the color ripple color is always grey (should be red with this sample)
- when user configure the  android_ripple={color: "#FFFFFF4F"} (With transparency) => the ripple color is blue. depending of transparency level, the color is sometimes grey ...
- I have also notice that a fast refresh doesn't refresh correctly the ripple color. (maybe it can help to identify the issue, or at least don't fall in that trap)

I checked with default react native `Pressable` these 3 use cases works fine.

### Steps to reproduce

Just define a Pressable with android_ripple={{color}}

```JS
<Pressable onPress={() => console.log('onPress')} android_ripple={{color: '#FF0000'}}>
         <Text>'Hello World'</Text>
</Pressable>
```

### Snack or a link to a repository

https://github.com/freeboub/bug-react-native-gesture-handler-android-ripple-color

### Gesture Handler version

2.21.2

### React Native version

0.76.2

### Platforms

Android

### JavaScript runtime

Hermes

### Workflow

Expo bare workflow

### Architecture

Fabric (New Architecture)

### Build type

Release mode

### Device

Android emulator

### Device model

reproduced on all android devices

### Acknowledgements

Yes

**Comment by freeboub** [comment]
I tried to debug the issue and I have an interesting point, This issue is reproducible only with new architecture enabled.
JS side looks good, but setRippleColor on kotlin side is never called

**Comment by freeboub** [comment]
Ok, I got the root cause, it looks like processColor should not be called with new architecture: 
doing this in Pressable.tsx fix the issue:
`rippleColor={android_ripple?.color ?? defaultRippleColor} `
But it doesn't work with old architecture 😭

**Comment by freeboub** [comment]
following patch can also be added to test the feature in the sample, but it breaks ios behavior, Not sure I should do it:
```diff
diff --git a/FabricExample/src/Navigator.tsx b/FabricExample/src/Navigator.tsx
index de9759f8e..d2d8227e0 100644
--- a/FabricExample/src/Navigator.tsx
+++ b/FabricExample/src/Navigator.tsx
@@ -1,7 +1,8 @@
 /* eslint-disable react-native/no-inline-styles */
 import React, { useEffect, useState } from 'react';
-import { View, Text, Pressable, BackHandler } from 'react-native';
-
+import { View, Text, BackHandler } from 'react-native';
+import { Pressable } from 'react-native-gesture-handler';
+import { COLORS } from './colors';
 export interface RouteInfo {
   component: React.ComponentType;
   title?: string;
@@ -27,6 +28,7 @@ const Button = (props: ButtonProps) => {
         height: 48,
         justifyContent: 'center',
       }}
+      android_ripple={{color: COLORS.NAVY}}
       onPress={() => {
         props.onPress();
       }}>
````

**Comment by ulyavrubel** [comment]
Same issue with RectButton component. Only with new architecture enabled
See expo snack https://snack.expo.dev/@uvrubel/rectbutton-android-ripple-color

**Comment by latekvo** [comment]
Please let me know if this PR fixes your issues: https://github.com/software-mansion/react-native-gesture-handler/pull/3369

**Comment by ulyavrubel** [comment]
@latekvo It's fixed in 2.23.1. Thank you!

**Comment by atulgpt17** [comment]
Hi @freeboub do you know where the conversion of 'transparent' to integer happening in new arch under Fabric? Asking as getting a crash of `ColorValue: the value must be a number or Object.` when using `react-native-gesture-handler` `TouchableWithoutFeedback` in my work project under new arch(though I am not able to repro this issue in a sample project)

**Comment by freeboub** [comment]
> Hi @freeboub do you know where the conversion of 'transparent' to integer happening in new arch under Fabric? Asking as getting a crash of `ColorValue: the value must be a number or Object.` when using `react-native-gesture-handler` `TouchableWithoutFeedback` in my work project under new arch(though I am not able to repro this issue in a sample project)

No, but check both PR, mine which was cancelled and the merged one. With this code you should be able to find all color conversations

**Comment by SudoPlz** [comment]
@atulgpt17 I attempted to create a fix for this on my PR https://github.com/software-mansion/react-native-gesture-handler/pull/3551 

Essentially passing 'transparent' as a string crashes on Fabric because it skips `processColor`, and the native side doesn’t handle strings directly. Switched it to the numeric value (0x00000000) so it works in both Fabric and non-Fabric. 🙏

I'd appreciate some eyes on my PR, thanks :D

**Comment by atulgpt17** [comment]
Hi @SudoPlz thanks for raising the PR, but in my case, I was missing the call of `com.facebook.react.defaults.DefaultNewArchitectureEntryPoint.load;` from my android custom application class which was not populating few globals in the JS `global` to enable the bridgeless mode and hence `FabricUIManager`
At the same time `isFabric` check in this repo which has implementation of `return !!global?.nativeFabricUIManager;` returns true in both the cases(when load is called, `global.RN$Bridgeless` is `true` and load is not called and `global.RN$Bridgeless` is `undefined`). 

I think check here should be updated, as some RN might prefer to run the application without bridgeless mode and in that case normalisation of `ColorValue` will not happen and will lead to crash

@freeboub do you agree here? I think we should also check `global.RN$Bridgeless === true`(if this is not true or undefined then isntead of picking `BridgelessUIManager` RN picks `PaperUIManager`) as well

Also @SudoPlz you PR made this conversion but ideally this conversion should be handled by the new fabric UI manager automatically so you PR should not be needed
